python/paddle/fluid/trainer.py

```diff
@@ -193,14 +253,18 @@ def _dist_transpile_if_necessary(self, optimize_ops, params_grads):
         current_endpoint = os.getenv("PADDLE_CURRENT_IP", "") + ":" + port
         # the unique trainer id, starting from 0, needed by trainer
         # only
-        trainer_id = int(os.getenv("PADDLE_TRAINER_ID", "0"))
+        self.trainer_id = int(os.getenv("PADDLE_TRAINER_ID", "0"))
+        self.chief = self.trainer_id == 0
```

Comment:
Why hard code `trainer_id` and `chief` ere equal to 0 ?

Reply:
We will default initialize ```trainer_id =0 ```  and ```chief = True ``` as default.
If run PaddlePaddle as local, there is only one trainer, its ```trainer_id ``` is 0, and it is the ```chief``` obviously.
If run PaddlePaddle as distribution, we will get ```PADDLE_TRAINER_ID ``` from ```env```, there will only have one trainer as the chief.